### DISCUSSION TITLE
question about multiple shells in spacemacs

### DISCUSSION THREAD

**Discussion by xyderos** [syl20bnr/spacemacs#15099] [Q&A]
Hi!

Is there a way to have multiple shells in spacemacs? 

For example, can I have in the 1st monitor a shell that isnt the same instance of the shell in the 2nd monitor?

Thanks!

**Comment by JAremko** [comment]
@xyderos  Well, if you need only 2 You can start `term` and `ansi-term` :smile:

**Comment by JAremko** [comment]
also you probably can start a terminal then rename it's buffer with `rename-buffer` and start a new one :thinking:  Actually sounds like a useful function to have. Maybe we have something like this :thinking:

**Comment by JAremko** [comment]
There is [MultiTerm package](https://www.emacswiki.org/emacs/MultiTerm) and we're [using it](https://github.com/syl20bnr/spacemacs/blob/develop/layers/+tools/shell/packages.el#L24) so it should be possible.

```elisp
      (spacemacs/set-leader-keys-for-major-mode 'term-mode
        "c" 'multi-term
        "C" 'term-char-mode
        "l" 'term-line-mode
        "n" 'multi-term-next
        "N" 'multi-term-prev
        "p" 'multi-term-prev)
```

**Comment by alexey0308** [comment]
@xyderos running `shell` with universal-argument allows to specify buffer name during creation:
`SPC-u M-x shell` then e.g. *shell2*.

**Comment by luyangliuable** [comment]
If you have multiple projects you have can different shell for each project. Try spc p $

**Comment by luyangliuable** [comment]
> @xyderos running `shell` with universal-argument allows to specify buffer name during creation: `SPC-u M-x shell` then e.g. _shell2_.

Wow thanks
